debug/gdbserver.py

```diff
@@ -1832,6 +1833,51 @@ def test(self):
         self.gdb.p("$pc=_start")
 
         self.exit()
+class CeaseStepiTest(ProgramTest):
+    """Test that we work correctly when the hart we're debugging ceases to
+    respond."""
+    def early_applicable(self):
+        return self.hart.support_cease
+
+    def test(self):
+        self.gdb.b("main")
+        output = self.gdb.c()
+        assertIn("Breakpoint", output)
+        assertIn("main", output)
+
+        self.gdb.p("$pc=cease")
+        self.gdb.stepi(wait=False)
```

Comment:
So the wait=False is introduced in order to be prepared for, and verify the presence of, asynchronous output that reports the available->unavailable change, whereas previously a stepi would be expected to complete silently in all cases.  Looks good to me.